#!/usr/bin/env python
PACKAGE = "livox_camera_calib"

from math import pi

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("voxel_size", double_t, 0, "Size of the voxel (m)", 1.0, 0.1, 5.0)
gen.add("down_sample_size", double_t, 0, "size of down sample point cloud (m)", 0.02, 0.01, 0.10)
gen.add("plane_min_points_size", int_t, 0, "min points for a valid plane", 15, 5, 100)
gen.add("plane_max_size", int_t, 0, "max planes to extract edges from", 8, 2, 10)
gen.add("ransac_dis_threshold", double_t, 0, "plane ransac distance threshold (m)", 0.02, 0.01, 0.2)
gen.add("line_min_dis_threshold", double_t, 0, "the smaller distance from fitted line to points (m)", 0.03, 0.01, 0.3)
gen.add("line_max_dis_threshold", double_t, 0, "the bigger distance from fitted line to points (m)", 0.06, 0.02, 0.6)
gen.add("normal_theta_min", double_t, 0, "min plane norm theta (degree)", 30.0, 0.0, 180.0)
gen.add("normal_theta_max", double_t, 0, "max plane norm theta (degree)", 150.0, 0.0, 180.0)

exit(gen.generate(PACKAGE, "livox_camera_calib", "Calib"))
